### PR TITLE
Add GitHub CI job to check Markdown links during PRs

### PR DESCRIPTION
**Pull request by joelhans**
##### Summary

Third step of #8990. This PR adds a GitHub CI workflow to check changed Markdown files for broken links. This check will ensure we don't merge PRs that contain broken links.

If we can implement #8991, #8994, and this PR, we should be ready to completely disconnect the old Netlify CI workflows.

I'm relatively new to GitHub Actions, so apologies in advance...

##### Component Name

.github
docs

##### Test Plan

This PR contains a broken link so reviewers can check its behavior and output. I'll remove it before merging.

### CODE REVIEW

File: .github/workflows/check-links.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+---
+name: Check Markdown links
```
Comment (prologic) [inline .github/workflows/check-links.yml:2]:
```suggestion
name: Docs
```

File: .github/workflows/check-links.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+on:
+  push:
+    branches:
+      - master
+  pull_request:
```
Comment (prologic) [inline .github/workflows/check-links.yml:7]:
```suggestion
  push:
    branches:
      - master
    paths:
      - '*.md'
  pull_request:
    paths:
      - '*.md'
```

File: .github/workflows/check-links.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+jobs:
+  markdown-link-check:
+    runs-on: ubuntu-latest
```
Comment (prologic) [inline .github/workflows/check-links.yml:10]:
```suggestion
  markdown-link-check:
    name: Broken Links
    runs-on: ubuntu-latest
```

File: .github/workflows/check-links.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@master
```
Comment (prologic) [inline .github/workflows/check-links.yml:13]:
```suggestion
        uses: actions/checkout@v2
```

File: registry/README.md [hunk 5]
```diff
@@ -15,6 +15,8 @@ features, you need to Sign In to Netdata cloud. By signing in, you opt in to let
 information described [here](#what-data-does-the-registry-store). You can still get the node menu, but not the cloud
 features, if you [run your own registry](#run-your-own-registry).
 
+[This is a broken link](/broken/link/README.md).
+
```
Comment (prologic) [inline registry/README.md:19]: I _assume_ you will remove this later in this PR once reviewed/approved :)

Comment (joelhans) [reply registry/README.md:19]: You read my mind :smile:

**Comment by joelhans** [comment]
@prologic I addressed your comments and removed the broken link. Thanks for the quick feedback!

Here's what the error output looks like, for reference:

```
 =========================> MARKDOWN LINK CHECK <=========================


FILE: registry/README.md
[✖] /broken/link/README.md → Status: 400 [Error: ENOENT: no such file or directory, access '/broken/link/README.md'] {

=========================================================================
```

**Comment by prologic** [comment]
> @prologic I addressed your comments and removed the broken link. Thanks for the quick feedback!

No worries! Sorry for the slow follow up ;/ I've been swamped with large volumes of notifs and some hard focus time to get things done. This happens from time-to-time :) bare with me! I'll re-review this today!

**Comment by joelhans** [comment]
@prologic Thank you very much! Let's take a day or two to make sure this definitely works as expected on new PRs and then we can move ahead with #8783.

**Comment by prologic** [comment]
> @prologic Thank you very much! Let's take a day or two to make sure this definitely works as expected on new PRs and then we can move ahead with #8783.

Sure! Sounds good to me. Please remind me when its time and just hit me up on Slack!

**Pull request merged**